### DISCUSSION TITLE
add some multiple pattern search in GUIDE.md

### DISCUSSION THREAD

**Discussion by jianhe-fun** [BurntSushi/ripgrep#2306] [General] [answered]
https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md
generally I use pipe when deal with multiple regex pattern. I know there is other way, but I don't know how to do it. 
If you search in GUIDE.md, there are just so many "and" so "and" key word search is useless. 
can you add some examples in  GUIDE.md about multiple regex pattern. It should include key word (multiple) So people can easily search it.   
reference: https://unix.stackexchange.com/questions/55359/how-to-run-grep-with-multiple-and-patterns
`ps aux | rg  'postgres.*wal.*'`
return
```
jian      148987  0.0  0.0 201184  5336 ?        Ss   Sep16   0:01 postgres: walwriter
jian      269356  0.0  0.0   8488  5916 pts/4    S+   10:27   0:00 rg postgres.*wal.*
```

**Comment by BurntSushi** [comment, marked as answer]
Your example doesn't use multiple patterns. I only see one. Can you give an actual example of what you want to see explained?

Are you talking about the `-f/--file` flag? Or things like `rg foo | rg bar`? The latter is "just" standard shell program composition.

**Reply by jianhe-fun** [reply]
Yes. `rg foo | rg bar` If I understand correctly, means match pattern1 foo and pattern2 bar. Obviously there is another way to express it in rg. But I don't know how.

**Reply by BurntSushi** [reply]
Why is there obviously some other way? There is not. That's what #875 is about. GNU grep doesn't have a way to do this other than `grep foo | grep bar` either. `git grep` does though.
